Mathlib/Algebra/FreeMonoid/Count.lean

```diff
@@ -27,7 +27,8 @@ def countP : FreeAddMonoid α →+ ℕ where
   map_add' := List.countP_append _
 
 theorem countP_of (x : α) : countP p (of x) = if p x = true then 1 else 0 := by
-  simp [countP, List.countP, List.countP.go]
+  change List.countP p [x] = _
```

Comment:
Should we have a simp(?) lemma that brings the LHS in this form?

Reply:
No, I don't think we should, unless someone actually adds the explicit conversion from FreeMonoid to List. As is, it is defeq abuse.

Reply:
See https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Mathlib.2FAlgebra.2FFreeMonoid.2FCount.2Elean/near/490048032

Reply:
Aah, thanks for that pointer to zulip.
In that case, I suggest we also leave a comment at the top of the file with the same comments as (or a pointer to) that zulip thread.